src/herder/HerderImpl.cpp

```diff
@@ -1608,16 +1608,18 @@ HerderImpl::restoreSCPState()
     ZoneScoped;
     // setup a sufficient state that we can participate in consensus
     auto const& lcl = mLedgerManager.getLastClosedLedgerHeader();
+
+    setTrackingSCPState(lcl.header.ledgerSeq, lcl.header.scpValue, false);
+
     if (!mApp.getConfig().FORCE_SCP &&
         lcl.header.ledgerSeq == LedgerManager::GENESIS_LEDGER_SEQ)
     {
         // if we're on genesis ledger, there is no point in claiming
         // that we're "in sync"
+        lostSync();
```

Comment:
I think we can do better than this in this commit:
looking at how the various states are used at this point, we can *remove* the `HERDER_TRACKING_LCL_STATE` state.

That way we get rid of the `bool` used by `setTrackingSCPState`.

Only difference will be:
when we startup we're "in sync" (tracking), unless we're on genesis ledger (out of sync, per this commit).

Other functional change is that `ApplicationImpl::getState()` can:
* if herder is "booting" --> application state is `APP_CREATED_STATE` (right now it lies with `APP_ACQUIRING_CONSENSUS_STATE`)
* if herder is not tracking --> application state is `APP_ACQUIRING_CONSENSUS_STATE`
* if herder is tracking --> use lm's state to determine state

difference with master is that on startup if "tracking lcl" but didn't see consensus yet, master claims "joining SCP" where as now it will say "Connected" (which is actually better). Comment above `APP_CONNECTED_STANDBY_STATE` should be updated to say "tracking" (as it doesn't mean "in sync").

That way after this commit we'll only have 2 modes for herder: tracking and "out of sync".

What do you think?

Reply:
> difference with master is that on startup if "tracking lcl" but didn't see consensus yet, master claims "joining SCP" where as now it will say "Connected" (which is actually better).

this is the case when FORCE_SCP is set to false. If it's set to true, LM is set to "synced" state, so the app will return "Synced!" before it hears from the network. On one hand, logically it makes sense to me, as starting the app with FORCE_SCP implies that it is in sync. On the other hand, we might have external dependencies on app's "synced" state (for example, in simulations). Besides tests, does anything else depend on it?

Reply:
@MonsieurNicolas so what do you think about "synced" being herder synced + lm synced + closed at least one ledger since startup? I know you also proposed using ledger age, but I think it's less accurate, because as soon as we start the app, ledger age will be pretty small, causing the app to go in sync.

Reply:
per our offline chat, I think we can just consider the node "in sync" on startup in this case. Maybe we can improve that signal later on when we improve how we trigger (don't set LM as synced on startup: the only way LM gets to sync state is by closing a ledger)